src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -202,6 +202,69 @@ public bool NoDownloadRequested(ConnectNewHeadersResult connectNewHeadersResult)
             }
+
+            /// <summary>
+            /// Initial setup for tests 18-20, 28.
+            /// Chain header tree setup. Initial chain has 4 headers.
+            /// SetUp:
+            ///                        =8d=9d=10d
+            ///                   6a=7a=8a=9a
+            /// GENESIS=1=2=3=4=5=
+            ///                   6b=7b=8b=9b
+            ///             3c=4c=5c
+            /// </summary>
+            /// <param name="initialChainSize"></param>
+            /// <param name="ctx">TestContext</param>
+            /// <param name="cht">ChainHeaderTree</param>
+            /// <param name="initialChainTip">Initial chain tip</param>
+            /// <param name="chainATip">out Chaind A tip</param>
+            /// <param name="chainDTip">out Chaind D tip</param>
+            /// <returns>List of BlockHeaders</returns>
+            internal IEnumerable<BlockHeader> SetupPeersForTest(int initialChainSize, TestContext ctx, ChainedHeaderTree cht, ChainedHeader initialChainTip, out ChainedHeader chainATip, out ChainedHeader chainDTip)
+            {
+                int peerAExtension = 4;
+                int peerBExtension = 4;
+                int peerCExtension = 3;
+                int peerDExtension = 3;
+
+                List<BlockHeader> initialchainHeaders = ctx.ChainedHeaderToList(initialChainTip, initialChainSize + 1);
+                chainATip = ctx.ExtendAChain(peerAExtension, initialChainTip); // i.e. (h1=h2=h3=h4=h5)=6a=7a=8a=9a
+                ChainedHeader chainBTip = ctx.ExtendAChain(peerBExtension, initialChainTip); // i.e. (h1=h2=h3=h4=h5)=6b=7b=8b=9b
+                ChainedHeader chainCTip = ctx.ExtendAChain(peerCExtension, initialChainTip.GetAncestor(2)); // i.e. (h1=h2)=3c=4c=5c
+                chainDTip = ctx.ExtendAChain(peerDExtension, chainATip.GetAncestor(7)); // i.e. ((h1=h2=h3=h4=h5)=6a=7a)=8d=9d=10d
+
+                List<BlockHeader> peerABlockHeaders = ctx.ChainedHeaderToList(chainATip, chainATip.Height);
+                List<BlockHeader> peerBBlockHeaders = ctx.ChainedHeaderToList(chainBTip, chainBTip.Height);
+                List<BlockHeader> peerCBlockHeaders = ctx.ChainedHeaderToList(chainCTip, chainCTip.Height);
+                List<BlockHeader> peerDBlockHeaders = ctx.ChainedHeaderToList(chainDTip, chainDTip.Height);
+
+                List<BlockHeader> listOfAllBlockHeaders = new List<BlockHeader>(initialchainHeaders);
+                listOfAllBlockHeaders.AddRange(peerABlockHeaders);
+                listOfAllBlockHeaders.AddRange(peerBBlockHeaders);
+                listOfAllBlockHeaders.AddRange(peerCBlockHeaders);
+                listOfAllBlockHeaders.AddRange(peerDBlockHeaders);
+                IEnumerable<BlockHeader> listOfAllUniqueBlockHeaders = listOfAllBlockHeaders.Distinct();
+
+                ConnectNewHeadersResult connectionPeerAResult = cht.ConnectNewHeaders(1, peerABlockHeaders);
```

Comment:
`ConnectNewHeadersResult connectionPeerAResult = ` is redundant here and on the following lines

Reply:
First peer ID is 0, not one.